src/psyclone/psyir/symbols/datatypes.py

```diff
@@ -740,8 +740,15 @@ def create(components):
         '''
         Creates a StructureType from the supplied list of properties.
 
-        :param components: the name, type and visibility of each component.
-        :type components: list of 4-tuples
+        :param components: the name, type, visibility and initial value (if
+                           any) of each component.
+        :type components: List[tuple[
+            str,
+            :py:class:`psyclone.psyir.symbols.DataType` |
+                 :py:class:`psyclone.psyir.symbols.DataTypeSymbol`,
+            :py:class:`psyclone.psyir.symbols.xxx`,
+            Optional[:py:class:`psyclone.psyir.symbols.Symbol.Visibility`]
+            ]]
```

Comment:
This doesn't seem right, Visibility and IntitialValue order is switched and InvitialValue should be a Optional[DataNode], not a symbol. You could take the docstring of the add method in this class, which is really that these components must match.

Reply:
Oh yes. That 'xxx' indicates I hadn't finished writing it! Hopefully fixed now.